### ISSUE TITLE
Quantity editing error

### ISSUE THREAD

**Issue by Moult** [IfcOpenShell/IfcOpenShell#2969]
If there are no cost items in the project, then the next line will still be passing a None cost_item. This breaks basic quantity editing.

 1. New IFC project. Create any object.
 2. Attempt to add a new Qto to that object with any value.
 3. Error :(

https://github.com/IfcOpenShell/IfcOpenShell/commit/4d34404e47f24d87cdea975c8c0e3e8d0f8f7eca#diff-3786dad2ddc914d25055c420a3f2e02ecbe2698b29b08ff882c2afc9f461b8a1R637

I'm also not convinced about this "temp fix" https://github.com/IfcOpenShell/IfcOpenShell/commit/cbe944d03086506fc1b12a9aa28545ffa932c4c0 - there is probably a better approach.

**Comment by theoryshaw** [comment]
experienced similar error.

```
Python: Traceback (most recent call last):
  File "C:\Users\Owner\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\blenderbim\bim\module\pset\operator.py", line 39, in execute
    IfcStore.execute_ifc_operator(self, context)
  File "C:\Users\Owner\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\blenderbim\bim\ifc.py", line 410, in execute_ifc_operator
    result = getattr(operator, "_execute")(context)
  File "C:\Users\Owner\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\blenderbim\bim\module\pset\operator.py", line 298, in _execute
    bpy.ops.bim.load_cost_item_quantities()
  File "C:\Program Files\Blender Foundation\Blender 3.3\3.3\scripts\modules\bpy\ops.py", line 113, in __call__
    ret = _op_call(self.idname_py(), None, kw)
RuntimeError: Error: Python: Traceback (most recent call last):
  File "C:\Users\Owner\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\blenderbim\bim\module\cost\operator.py", line 516, in execute
    core.load_cost_item_quantities(tool.Cost)
  File "C:\Users\Owner\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\blenderbim\core\cost.py", line 112, in load_cost_item_quantities
    cost.load_cost_item_quantities()
  File "C:\Users\Owner\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\blenderbim\tool\cost.py", line 638, in load_cost_item_quantities
    tool.Cost.load_cost_item_quantity_assignments(cost_item, related_object_type="PRODUCT")
  File "C:\Users\Owner\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\blenderbim\tool\cost.py", line 202, in load_cost_item_quantity_assignments
    create_list_items(props.cost_item_products, cost_item, is_deep)
  File "C:\Users\Owner\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\blenderbim\tool\cost.py", line 187, in create_list_items
    products = cls.get_cost_item_assignments(cost_item, filter_by_type=related_object_type, is_deep=False)
  File "C:\Users\Owner\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\blenderbim\tool\cost.py", line 393, in get_cost_item_assignments
    return ifcopenshell.util.cost.get_cost_item_assignments(
  File "C:\Users\Owner\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\blenderbim\libs\site\packages\ifcopenshell\util\cost.py", line 212, in get_cost_item_assignments
    return get_cost_assignments_by_type(cost_item, filter_by_type)
  File "C:\Users\Owner\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\blenderbim\libs\site\packages\ifcopenshell\util\cost.py", line 205, in get_cost_assignments_by_type
    for r in cost_item.Controls or []
AttributeError: 'NoneType' object has no attribute 'Controls'
Location: C:\Program Files\Blender Foundation\Blender 3.3\3.3\scripts\modules\bpy\ops.py:113

```

**Comment by myoualid** [comment]
This commit should fix it:  ed4ba4d0d3d5d3ff95e04a85ebce3df244cdda8c - a few things were improved along the bug fix
